### DISCUSSION TITLE
Support filter by regex?

### DISCUSSION THREAD

**Discussion by bluelovers** [supabase/supabase#1822] [Questions] [answered]
does supabase support filter by regex?

**Comment by steve-chavez** [comment, marked as answer]
Hey @bluelovers,

We don't expose regex directly through the libraries or the REST API because it's not safe to do so - see [ReDoS](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS). In the past I've managed to make postgres hang on an infinite loop through [a regex](https://github.com/PostgREST/postgrest/issues/539#issuecomment-255616248). 

That being said, you can create a SQL function and use a regex inside it, making sure you do some validation on the input to prevent any abuse. Then you can call that function through [supabase.rpc](https://supabase.io/docs/reference/javascript/rpc).

Before doing that though, I'd recommend to look at the [like](https://supabase.io/docs/reference/javascript/like) and [textSearch](https://supabase.io/docs/reference/javascript/textsearch) filters. Are you sure you cannot search your string with those?

**Reply by bluelovers** [reply]
does like and textSearch can do `(a|b)(c|d)[def]` this?

**Reply by steve-chavez** [reply]
That one looks like a legitimate use case for a regex. You can do this for now:

```sql
-- assuming there's a projects table
create or replace function projects_regex(regex text) returns setof projects as $$
  select * from projects where name ~ regex;
$$ language sql;
```

Then through rpc:

```js
const { data, error } = await supabase
  .rpc('projects_regex', { regex: '(a|b)(c|d)[def]' })
```
rpc supports calling [filters](https://supabase.io/docs/reference/javascript/rpc#with-filters), ordering, even `select`, so it's pretty flexible. You could also limit the input size on the function if you want to be extra safe.
